File: src/Unit-2/DoThis/ChartDataHelper.cs
using System;
using System.Linq;
using System.Windows.Forms.DataVisualization.Charting;
using Akka.Util;

namespace ChartApp
{
    /// <summary>
    /// Helper class for creating random data for chart plots
    /// </summary>
    public static class ChartDataHelper
    {
        public static Series RandomSeries(string seriesName, SeriesChartType type = SeriesChartType.Line, int points = 100)
        {
            var series = new Series(seriesName) {ChartType = type};
            foreach (var i in Enumerable.Range(0, points))
            {
                var rng = ThreadLocalRandom.Current.NextDouble();
                series.Points.Add(new DataPoint(i, 2.0*Math.Sin(rng) + Math.Sin(rng/4.5)));
            }
            series.BorderWidth = 3;
            return series;
        }
    }
}

File: src/Unit-2/lesson1/FSharp/Completed/ChartDataHelper.fs
module ChartDataHelper

open System.Windows.Forms.DataVisualization.Charting
open Akka.Util

let randomSeries (seriesName: string) (seriesType: SeriesChartType option) (points: int option) =
    let seriesChartType = defaultArg seriesType SeriesChartType.Line
    let seriesPoints = defaultArg points 100 
    let series = new Series(seriesName, ChartType = seriesChartType);
    [0..seriesPoints]
    |> List.iter (fun i -> 
            let rng = ThreadLocalRandom.Current.NextDouble ()
            series.Points.Add(new DataPoint(float i, float (2.0 * sin(rng) + sin(rng / 4.5))))
            series.BorderWidth <- 3)
    series
